caikit/runtime/work_management/train_executors.py

```diff
@@ -0,0 +1,275 @@
+                # task threw exception
+                self._worker.get_or_throw()
+
+        # Handle errors as CaikitRuntime errors with appropriate error codes
+        except CaikitRuntimeException as e:
+            log.warning(
+                {
+                    "log_code": "<RUN555430380W>",
+                    "message": e.message,
+                    "error_id": e.id,
+                    **e.metadata,
+                }
+            )
+            raise e
+        except (TypeError, ValueError) as e:
+            log.warning(
+                {
+                    "log_code": "<RUN868639039W>",
+                    "message": repr(e),
+                    "stack_trace": traceback.format_exc(),
+                }
+            )
+            raise CaikitRuntimeException(
+                StatusCode.INVALID_ARGUMENT,
+                f"Exception raised during training. This may be a problem with your input: {e}",
+            ) from e
+        except Exception as e:
+            log.warning(
+                {
+                    "log_code": "<RUN490967039W>",
+                    "message": repr(e),
+                    "stack_trace": traceback.format_exc(),
+                }
+            )
+            raise CaikitRuntimeException(
+                StatusCode.INTERNAL,
+                f"Exception raised during training: {e}",
+            ) from e
+
+    def cancel(self) -> None:
+        """Function to abort train and save operation on the executor"""
+        self._worker.destroy()
+
+
+class SubProcessTrainSaveExecutor(TrainSaveExecutorBase):
+    class _ErrorCaptureProcess(multiprocessing.get_context("fork").Process):
+        """This class wraps a Process and keeps track of any errors that occur
+        during execution
+
+        NOTE: We explicitly use "fork" here for two reasons:
+            1. It's faster
+            2. Due to the auto-generated classes with stream sources, "spawn"
+            can result in missing classes since it performs a full re-import,
+            but does not regenerate the service APIs
+        """
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.error = None
+
+        def set_args(self, *args, **kwargs):
+            self._args = args
+            self._kwargs = kwargs
+
+        def run(self, *args, **kwargs):
+            try:
+                return super().run(*args, **kwargs)
+
+            # Catch any errors thrown within a subprocess so that they can be
+            # forwarded to the parent
+            # pylint: disable=broad-exception-caught
+            except Exception as err:
+                self.error = err
+
+    def __init__(self, event) -> None:
+
+        self._worker = self._ErrorCaptureProcess(
+            target=TrainSaveExecutorBase.train_and_save,
+        )
+
+        self.__proc_id = self._worker.pid
+        self.__event = event
+
+    def __del__(self):
+        self.cancel()
```

Comment:
Let's add a docstring here to explicitly say that this is NOT how cancelation should be invoked, but that it's here to make sure cleanup happens